client/db/src/lib.rs

```diff
@@ -1345,8 +1345,7 @@ impl<Block: BlockT> Backend<Block> {
 
 			let parent_hash = *pending_block.header.parent_hash();
 			let number = *pending_block.header.number();
-			let existing_header =
-				number <= best_num && self.blockchain.header(BlockId::hash(hash))?.is_some();
+			let existing_header = self.blockchain.header(BlockId::hash(hash))?.is_some();
```

Comment:
We could keep the check if we combine it with this here: https://github.com/paritytech/substrate/pull/11716/files#diff-d631f4b5aa83ad9f2088c7d13236ad51b6a4ffdb870927f071532d9904bc2c5fR264

Aka that we not compare against `best_num` and instead compare against `highest_leaf`.

Reply:
We could check if parent is a known leaf instead.